aria/modeling/service_template.py

```diff
@@ -549,9 +549,13 @@ def as_raw(self):
             ('requirement_templates', formatting.as_raw_list(self.requirement_templates))))
 
     def instantiate(self, container):
-        context = ConsumptionContext.get_thread_local()
         from . import models
-        name = context.modeling.generate_node_id(self.name)
+        if self.nodes:
+            latest_node_index = self.nodes[-1].name.rsplit('_', 1)[1]
```

Comment:
is the nodes list guaranteed to be ordered?
if not then we need to find the max

either way, i'd change `latest` to `highest`